test/compile_errors.zig

```diff
@@ -3877,7 +3877,7 @@ pub fn addCases(cases: *tests.CompileErrorContext) void {
         \\fn f(a: noreturn) void {}
         \\export fn entry() void { f(); }
     , &[_][]const u8{
-        "tmp.zig:1:9: error: parameter of type 'noreturn' not allowed",
+        "tmp.zig:1:9: error: parameter of NoReturn type 'noreturn' not allowed",
```

Comment:
The error message is a bit weird, `NoReturn type 'noreturn'` is redundant and users may wonder why `NoReturn` is capitalized.

Reply:
It's because in case of opaque types, they can be named. Then it looks like
`error: parameter of Opaque type 'MyType' not allowed`
which explains why `MyType` is not allowed.

Reply:
If it only makes sense for opaque types then show that only for opaque types, no?